powerlevel9k.zsh-theme

```diff
@@ -826,18 +826,21 @@ prompt_dir() {
       parts[1,-2]=()
     ;;
     truncate_to_first_and_last)
-      if (( POWERLEVEL9K_SHORTEN_DIR_LENGTH > 0 )); then
-        local -i i=$(( POWERLEVEL9K_SHORTEN_DIR_LENGTH + 1 ))
+      if (( shortenlen > 0 )); then
+        local -i i=$(( shortenlen + 1 ))
         [[ $p == /* ]] && (( ++i ))
-        for (( ; i <= $#parts - POWERLEVEL9K_SHORTEN_DIR_LENGTH; ++i )); do
+        for (( ; i <= $#parts - shortenlen; ++i )); do
           parts[i]=$'\0'
         done
       fi
     ;;
     truncate_to_unique)
-      local -i i=2 n=1
+      local -i i=2 n=1 d=0
```

Comment:
Let's move `d` above `case`.

```zsh
local -i d=${POWERLEVEL9K_SHORTEN_DELIMITER_LENGTH:-$#POWERLEVEL9K_SHORTEN_DELIMITER}
```

And then use it in all strategies where it makes sense. I think it would involve replacing `1` with `d` [here](https://github.com/romkatv/powerlevel10k/blob/bbebf516abeb55c96ad6e83478581b1230df6ec2/powerlevel9k.zsh-theme#L776) and [here](https://github.com/romkatv/powerlevel10k/blob/bbebf516abeb55c96ad6e83478581b1230df6ec2/powerlevel9k.zsh-theme#L817).